### PR TITLE
:lipstick: ContactList --- Improve exceptions

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
1. Add messages to the exceptions
2. `get` specifies the issue in the message (negative, or too big). 
3. Add tests for exception messages to tests

### Testing
:+1:

### CODE REVIEW

File: src/test/java/ContactListTest.java [hunk 1]
```diff
@@ -211,12 +224,14 @@ void get_validIndex_returnsCorrectFriend() {
 
                 @Test
                 void get_negativeIndex_throwsIndexOutOfBoundsException() {
-                    assertThrows(IndexOutOfBoundsException.class, () -> classUnderTest.get(-1));
+                    Exception e = assertThrows(IndexOutOfBoundsException.class, () -> classUnderTest.get(-1));
+                    assertEquals("Cannot index with negative indices.", e.getMessage());
```
Comment (jameshughes89) [inline src/test/java/ContactListTest.java:228]: Upon further thought, I am not sure the messages are a great thing to check. Exception, yeah, but message, idk. 

E.g., if the interface says some method should throw an exception, then yeah, check for the exception. But the interface doesn't say what the message should be. 

Am I crazy?

**Comment by jameshughes89** [comment]
I removed them.

File: src/main/java/ContactList.java [hunk 2]
```diff
@@ -95,11 +95,11 @@ private int find(Friend friend) {
      *
      * @param friend Friend object to find the index of within the collection.
      * @return Index of the Friend object.
-     * @throws NoSuchElementException If no equal Friend object exists within the collection, throw an exception.
+     * @throws NoSuchElementException Throw if no equal Friend object exists within the collection, throw an exception.
```
Comment (twentylemon) [inline src/main/java/ContactList.java:98]: `@throws X throw if...` reads oddly. I prefer the prior text here.

File: src/test/java/ContactListTest.java (hunk 1 above)
Comment (jameshughes89) [reply src/test/java/ContactListTest.java]: checks out. 

Glad I cut them.

File: src/main/java/ContactList.java [hunk 3]
```diff
@@ -95,11 +95,11 @@ private int find(Friend friend) {
...
      */
     public int indexOf(Friend friend) {
         if (!contains(friend)) {
-            throw new NoSuchElementException();
+            throw new NoSuchElementException("Friend does not exist within ContactList.");
```
Comment (twentylemon) [inline src/main/java/ContactList.java:102]: It's typical to include as much information as reasonably possible in your exception messages. They only occur on failures, so spending some extra time generating a good message is cheap enough. I'd probably just do:
```java
throw new NoSuchElementException(Objects.toString(friend));
```
The `NoSuchElementException` already tells you that something does not exist, and the stack trace will tell you it's a `ContactList`. The value is the only thing missing.

Comment (jameshughes89) [reply src/main/java/ContactList.java]: Switched the index out of bouts exception to also have the index in the mesage

**Review by twentylemon — APPROVED** [review]

**Pull request merged**